gnucash/import-export/aqb/gnc-flicker-gui.c

```diff
@@ -0,0 +1,408 @@
+                bitarray[i+1][1] = bits[j][1];
+                bitarray[i+1][2] = bits[j][2];
+                bitarray[i+1][3] = bits[j][3];
+                bitarray[i+1][4] = bits[j][4];
+			}
+		}
+    }
+    g_free (arr1);
+    g_free (arr2);
+
+    return code;
+}
+
+/* A clock is started here and called up again when the "Delay" value is changed */
+gboolean
+time_handler (GtkWidget *widget)
+{
+//	g_print("[Aufruf]: %s \n", __FUNCTION__ );
+
+    /* Änderung der Wartezeit */
+    if (flickerdraw.change_interval)
+    {
+        g_source_remove (flickerdraw.interval);
+        flickerdraw.interval = g_timeout_add (flickerdraw.delay, (GSourceFunc) time_handler, (gpointer) widget);
+        flickerdraw.change_interval = FALSE;
+        return FALSE;
+    }
+    gtk_widget_queue_draw (widget);
+
+    return TRUE;
+}
+
+/* Show the colored triangle as a pointer for the position of the TAN generator */
+void
+doMarkerDrawing (cairo_t *cr)
+{
+	    guint pos1;
+	    guint pos2;
+
+	    /* Initialize the drawing area to black */
+	    cairo_set_source_rgb (cr, 0, 0, 0);
+	    cairo_paint (cr);
+
+	    cairo_set_source_rgb (cr, 0.9, 0.1, 0.1);
+	    /* Dreieck links */
+	    pos1 = flickerdraw.x_drawpos+flickerdraw.barwidth/2;
+	    cairo_move_to(cr, pos1, 20);
+	    cairo_line_to(cr, pos1+10, 2);
+	    cairo_line_to(cr, pos1-10, 2);
+	    cairo_close_path(cr);
+	    cairo_stroke_preserve(cr);
+	    cairo_fill(cr);
+
+	    /* Dreieck rechts */
+	    pos2 = flickerdraw.x_drawpos+4*flickerdraw.margin+4*flickerdraw.barwidth+flickerdraw.barwidth/2;
+	    cairo_move_to(cr, pos2, 20);
+	    cairo_line_to(cr, pos2+10, 2);
+	    cairo_line_to(cr, pos2-10, 2);
+	    cairo_close_path(cr);
+	    cairo_stroke_preserve(cr);
+	    cairo_fill(cr);
+}
+
+/* display the flicker graphic in the drawing area */
+void
+doFlickerDrawing (GtkWidget *widget, cairo_t *cr)
+{
+	guint i;
+
+    /* Initialize the drawing area to black */
+    cairo_set_source_rgb (cr, 0, 0, 0);
+    cairo_paint (cr);
+
+    bitarray[flickerdraw.halfbyteid][0] = flickerdraw.clock;
+
+    /* die Flickeranzeige immer in der Mitte der Drawingarea ausrichten */
+    flickerdraw.width = gtk_widget_get_allocated_width (widget);
+    flickerdraw.x_drawpos = (flickerdraw.width - 4*flickerdraw.margin - 5*flickerdraw.barwidth)/2;		/* Startposition des ersten Balkens */
+
+    for ( i = 0; i <= 4; i++ ) {
+    	if ( bitarray[flickerdraw.halfbyteid][i] == TRUE ) {
+    		cairo_set_source_rgb(cr, 1, 1, 1); // white
+    	}
+    	else {
+    		cairo_set_source_rgb(cr, 0, 0, 0); // black
+    	}
+    	flickerdraw.x_barpos = flickerdraw.x_drawpos + i*flickerdraw.margin + i*flickerdraw.barwidth;
+    	cairo_rectangle (cr, flickerdraw.x_barpos, flickerdraw.y_barpos, flickerdraw.barwidth, flickerdraw.barheight);
+    	cairo_fill (cr);
+    }
+
+    /* Each flicker point is drawn twice. Once with clock = 1 and once with clock = 0 */
+    if ( flickerdraw.clock == 0 ) {
```

Comment:
It's poor style to say `if (foo == TRUE)` or `if (foo == FALSE)`  or the 1  and 0 equivalents when `foo` is a boolean type. Use `if (foo)` and `if (!foo)`.